### PR TITLE
Redirect desktop client when 426 response is received

### LINKED ISSUE #9660: Blocking older clients - Understand "426 Upgrade Required" on API calls
With an upcoming server change, it will be possible to enforce a minimum client version:
* https://github.com/nextcloud/spreed/pull/9632

This requires all future clients to be able to understand `426 Upgrade Required`. Ideally they should:
* Show a note to the user that the client is too old and no longer supported by the server.
* Have a "button of action" allowing the user to get to the update as quick as possible (link to iOS Appstore / Google Play Store / Desktop download page)
* Clients should pause making further requests to not DoS attack the server in an attempt to get a room list, etc.

- [x] API https://github.com/nextcloud/spreed/pull/9632
- [x] iOS https://github.com/nextcloud/talk-ios/issues/1254
- [x] Android https://github.com/nextcloud/talk-android/issues/3061
- [x] Desktop https://github.com/nextcloud/talk-desktop/issues/209

### PR DESCRIPTION
**Pull request by Antreesy**
### ☑️ Resolves

* Fix nextcloud/talk-desktop#209
* Fix #9660 
* Wording is aligned with mobile clients

### 🖼️ Screenshots

| 🏡 After |
|---|
| image |

### 🚧 Tasks

- [ ] Visual check
- [ ] Code review

To test: adjust constant to be higher than current Talk Desktop version (0.9.0):

https://github.com/nextcloud/spreed/blob/67c22f97d7d9a10fe085f7bc65ee4f0c23c9e652/lib/Middleware/CanUseTalkMiddleware.php#L53

### 🏁 Checklist

- [x] ⛑️ Tests (unit and/or integration) are included or not required
- [x] 📘 API documentation in `docs/` has been updated or is not required
- [x] 📗 User documentation in https://github.com/nextcloud/documentation/tree/master/user_manual/talk has been updated or is not required
- [x] 🔖 Capability is added or not needed

### CODE REVIEW

**Review by ShGKme — CHANGES_REQUESTED** [review]
I understand that the solution in this PR implements approach used for the talk hash invalidation. But I'd prefer to have a different approach here and refactor talk hash later.

cc @danxuliu for a second opinion here

---

I'm not sure it should be done in the Talk web frontend in general. It's a large part that is not relevant for the web frontend.

We can do it directly in the desktop client:
1. Check for the status in the http response (any response)
  - Simple way - add interceptor directly in the place, where we need to react on it. Aka in some root component add `axios.interceptors.response.use`
  - Better in my opinion way in my opinion - add 426 response handler in the http client to avoid http client abstraction leak here
- Add a new route with new view by [addRoute](https://router.vuejs.org/api/interfaces/Router.html#addRoute)
- Show the message in the desktop frontend and make the same redirect as in this PR

File: src/views/UpgradeRequiredView.vue [hunk 1]
```diff
@@ -0,0 +1,42 @@
+<template>
+	<div id="emptycontent">
+		<div id="emptycontent-icon" class="icon-search" />
+		<h2>{{ t('spreed', 'The app is too old and no longer supported by this server.') }}</h2>
```
Comment (nickvergessen) [inline src/views/UpgradeRequiredView.vue:4]:
```suggestion
		<h2>{{ t('spreed', 'The client version is too old and no longer supported by this server.') }}</h2>
```

File: src/store/talkHashStore.js [hunk 2]
```diff
@@ -96,12 +108,19 @@ const actions = {
...
-	checkMaintenanceMode(context, response) {
+	checkForMaintenanceOrUpgrade(context, response) {
 		if (response?.status === 503 && !context.state.maintenanceWarningToast) {
 			context.commit('setMaintenanceWarningToast', showError(t('spreed', 'Nextcloud is in maintenance mode, please reload the page'), {
 				timeout: TOAST_PERMANENT_TIMEOUT,
 			}))
 		}
+
+		if (response?.status === 426 && !context.state.upgradeWarningToast && IS_DESKTOP) {
+			context.commit('setUpgradeWarningToast', showError(t('spreed', 'The app is too old and no longer supported by this server. Update is required.'), {
```
Comment (nickvergessen) [inline src/store/talkHashStore.js:119]:
```suggestion
			context.commit('setUpgradeWarningToast', showError(t('spreed', 'The client version is too old and no longer supported by this server. Update is required.'), {
```

**Comment by Antreesy** [comment]
Closed in favor of https://github.com/nextcloud/talk-desktop/pull/289

**Comment by danxuliu** [comment]
> I understand that the solution in this PR implements approach used for the talk hash invalidation. But I'd prefer to have a different approach here and refactor talk hash later.
> 
> cc @danxuliu for a second opinion here

I am late to the party... but I fully agree :-)

**Pull request closed without merging**